gapis/config/config.go

```diff
@@ -16,25 +16,53 @@
 package config
 
 const (
-	DebugReplay                = false
-	DebugReplayBuilder         = false
+	// DebugReplay activates various debug logs related to replay.
+	DebugReplay = false
+
+	// DebugReplayBuilder activates various debug logs and checks related to the
+	// creation of a replay payload.
+	DebugReplayBuilder = false
+
+	// DisableDeadCodeElimination prevents the early computation of the
+	// dependency graph.
 	DisableDeadCodeElimination = false
-	DeadSubCmdElimination      = false
-	DebugDeadCodeElimination   = false
-	DebugDependencyGraph       = false
-	DumpReplayProfile          = false
-	DumpValidationTrace        = true
-	AllInitialCommandsLive     = false
-	LogExtrasInTransforms      = false // Logs all commands' extras together with transforms
-	LogMemoryInExtras          = false // Logs all commands' read/write memory observation together with extras
-	// Logs all mappings at the end of the replay from original trace
-	// handles to replay client handles (if handles are reused in the trace
-	// it will only print the last mapping).  Only works for Vulkan.
-	LogMappingsToFile        = false
-	LogTransformsToFile      = false
-	LogTransformsToCapture   = false
-	LogInitialCmdsIssues     = false
-	LogInitialCmdsToCapture  = false
-	SeparateMutateStates     = false
-	CheckRebuiltStateMatches = false
+
+	// DeadSubCmdElimination prevents the elimination of subcommands in DCE.
+	DeadSubCmdElimination = false
+
+	// DebugDeadCodeElimination activates various debug logs related to DCE.
+	DebugDeadCodeElimination = false
+
+	// DumpReplayProfile dumps the perfetto trace of a replay profile.
+	DumpReplayProfile = false
+
+	// DumpValidationTrace dumps the perfetto trace of a validation profile.
+	DumpValidationTrace = false
+
+	// AllInitialCommandsLive forces all initial commands to be considered as
+	// live when computing DCE.
+	AllInitialCommandsLive = false
+
+	// LogExtrasInTransforms logs all commands' extras together with transforms.
+	LogExtrasInTransforms = false
+
+	// LogMemoryInExtras logs all commands' read/write memory observation
+	// together with extras.
+	LogMemoryInExtras = false
+
+	// LogMappingsToFile logs all mappings at the end of the replay from
+	// original trace handles to replay client handles (if handles are reused in
+	// the trace it will only print the last mapping).  Only works for Vulkan.
+	LogMappingsToFile = false
+
+	// LogTransformsToFile logs all commands seen by each transform into a
+	// separate file for each transform.
+	LogTransformsToFile = false
+
+	// LogTransformsToCapture creates a gfxtrace that stores the commands
+	// obtained after each transformation has been applied.
```

Comment:
(1) From my understanding, this does not include the initial state setup. This may be worth pointing out as a note?
(2) Also, there are some setup/tear-down commands included. Maybe worth pointing out similarly?
(3) nit: "after all transformations have been applied"

E.g.
LogTransformsToCapture creates a gfxtrace that stores the commands after all transformations have been applied.
Note that transformations introduce additional commands for [not sure exactly, allocate memory? preserve state?].
Note also that for mid-execution captures, the resulting trace will not include the initial state setup (see `LogInitialCmdsToCapture`). In general it is therefore an invalid trace and cannot be replayed.

Reply:
(1) I believe it does, I just tried it to confirm. It may depend on the way you reach `gapis/api/vulkan/replay.go:API.Replay()`, but I think most if not all of the time you'll have the initial commands by then.

(2) The whole point of transforms is to add/remove commands (except maybe for `transform_file_log`), so I think we can assume developers will be aware that transforms have added/removed commands.

Transforms add commands for various reasons. To take an example, transform_read_framebuffer will add the necessary commands to read the framebuffer at a given point in the trace. These are not really documented apart from (1) being stored in a file name like `gapis/api/vulkan/transform_*.go`, and (2) a small comment at the top of the file, e.g.:

```
// readFramebuffer implements a transform that adds the necessary
// Vulkan commands to retrieve a framebuffer attachement.
type readFramebuffer struct {
```

(3) Fixed, thanks!